### DISCUSSION TITLE
Enable modules in tenants

### DISCUSSION THREAD

**Discussion by intellimedhu** [OrchardCMS/OrchardCore#8962] [Q&A]
Do you have any idea how to enable some features in a lot of tenants? I have ca. 100 ones. It would take a long to enable those manually.

**Comment by jtkech** [comment]
In any config source, e.g. the `appsettings.json`, use the `Features` array config value just under the `OrchardCore` section (not under a child section specific to a tenant) so that it is applied to all tenants, the features will be always enabled and for all tenants.

Note: You need to take care of the dependencies of your modules / features that need to be also included in the array.

    {
        "OrchardCore": {
        ...
        "Features": [ "ModuleName1", "ModuleName2", ... ]
        ...
        }
    }

**Reply by intellimedhu** [reply]
Thanks!

**Reply by intellimedhu** [reply]
Any idea how to execute receipes on these tenants?

**Reply by jtkech** [reply]
@intellimedhu 

Go to the admin, there is a page to list the existing recipe and to be able to run them

That said we only list the ones that are not **setup** recipe, for custom needs you will have to create your owns

**Reply by Piedone** [reply]
Is this documented somewhere @jtkech? Couldn't find it.

**Reply by jptissot** [reply]
@intellimedhu I started working on a module that allows you to run some Json (recipe) for a number of tenants. It would be trivial to allow running recipes (zip) as well. See: https://github.com/StatCan/StatCan.OrchardCore/pull/207 https://github.com/StatCan/StatCan.OrchardCore/pull/207/files#diff-5805dbfea15427ad8ab5ead0ff4d0871a7da99ebb61bf316148634da582fb607R60

**Reply by jptissot** [reply]
> 
> 
> Is this documented somewhere @jtkech? Couldn't find it.

This is for every tenant. Correct ?

**Reply by jtkech** [reply]
@Piedone 

> Is this documented somewhere @jtkech? Couldn't find it.

I don't think so ;)

Would need to be part of the tenant pre-configuration doc https://docs.orchardcore.net/en/dev/docs/reference/core/Configuration/#tenant-preconfiguration

@jptissot 

> This is for every tenant. Correct ?

Yes, can be specific to a tenant if under a child tenant specific section, or for all if just under the `OrchardCore` section
